### PR TITLE
FEATURE(rawx): Add `openio_rawx_state`

### PR DESCRIPTION
**Pull request by cdelgehier**
##### SUMMARY

This PR allows to remove an instance rawx
`openio_rawx_state` can be `present` (default) or `absent`

 ##### IMPACT
N/A

 ##### ADDITIONAL INFORMATION

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -38,6 +38,7 @@ Rawx is the storage service and is implemented as an apache webdav repository mo
 | `openio_rawx_provision_only` | `false` | Provision only without restarting services |
 | `openio_rawx_serviceid` | `"0"` | ID in gridinit |
 | `openio_rawx_slots` | `[rawx]` | The service's slot in conscience |
+| `openio_rawx_state` | `present` | If `absent` the configuration will be removed and service will be stop |
```
Comment (vincent-legoll) [inline README.md]: stopped

**Comment by vincent-legoll** [comment]
LGTM

File: tasks/main.yml [hunk 2]
```diff
@@ -16,92 +16,86 @@
...
-    - path: "/etc/oio/sds/{{ openio_rawx_namespace }}/watch"
-    - path: "/var/log/oio/sds/{{ openio_rawx_namespace }}/{{ openio_rawx_servicename }}"
-      owner: "{{ syslog_user }}"
-      mode: "0770"
-  tags: configure
+- name: Set properties
+  set_fact:
+    _rawx:
+      volume: "{{ item.1.volume }}"  # mandatory
+      name: "{{ openio_rawx_type }}-{{ item.1.id | default(item.0) }}"  # calculated
```
Comment (vincent-legoll) [inline tasks/main.yml:23]: `-{{ item.1.id | default(item.0.id) }}` ?

File: tasks/main.yml [hunk 3]
```diff
@@ -16,92 +16,86 @@
...
+      id: "{{ item.1.id | default(item.0) }}"  # mandatory
```
Comment (vincent-legoll) [inline tasks/main.yml:24]: same

File: tasks/main.yml [hunk 4]
```diff
@@ -16,92 +16,86 @@
...
+  when:
+    - not openio_rawx_provision_only
+    - need_reload | default(false)
 
 - name: "restart rawx to apply the new configuration"
-  shell: |
-    gridinit_cmd reload
-    gridinit_cmd restart  {{ openio_rawx_namespace }}-{{ openio_rawx_servicename }}
-  register: _restart_rawx
+  shell: gridinit_cmd restart  {{ openio_rawx_namespace }}-{{ openio_rawx_servicename }}
```
Comment (vincent-legoll) [inline tasks/main.yml:70]: shell => command & double space

File: tasks/offline.yml [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+    mode: 0640
+  with_items:
+    - src: "gridinit_rawx.conf.j2"
+      dest: "{{ openio_rawx_gridinit_dir }}/{{ openio_rawx_gridinit_file_prefix }}\
+        {{ openio_rawx_servicename }}.conf"
+  register: _rawx_conf
+
+- block:
+    - name: offline - stop rawx to apply the new configuration
+      shell: gridinit_cmd stop  {{ openio_rawx_namespace }}-{{ openio_rawx_servicename }}
```
Comment (vincent-legoll) [inline tasks/offline.yml:26]: shell => command & double space

**Review by vincent-legoll — CHANGES_REQUESTED** [review]
LGTM otherwise

File: tasks/main.yml (hunk 2 above)
Comment (cdelgehier) [reply tasks/main.yml:23]: item.0 doesn't have any key 'id'

**Pull request merged**